core/src/test/java/apoc/meta/MetaTest.java

```diff
@@ -931,4 +948,42 @@ public void testNodeTypePropertiesEquivalenceTypeMapping() throws Exception {
+                "(p)-[:VIEWED]->(m), (p)-[:BOUGHT{quantity: 10}]->(pr)");
+        Set<Map<String, Object>> expectedResult = new HashSet<>();
+        expectedResult.add(MapUtil.map("other",List.of(),"count",0L,"existence",false,"index",false,"label","BOUGHT","right",0L,"type","INTEGER","sample",null,"rightCount",0L,"leftCount",0L,"array",false,"left",0L,"unique",false,"property","quantity","elementType","relationship","otherLabels",List.of()));
+        expectedResult.add(MapUtil.map("other",List.of(),"count",0L,"existence",false,"index",false,"label","Product","right",0L,"type","STRING","sample",null,"rightCount",0L,"leftCount",0L,"array",false,"left",0L,"unique",false,"property","name","elementType","node","otherLabels",List.of()));
+        expectedResult.add(MapUtil.map("other",List.of("Product"),"count",1L,"existence",false,"index",false,"label","BOUGHT","right",1L,"type","RELATIONSHIP","sample",null,"rightCount",1L,"leftCount",1L,"array",false,"left",1L,"unique",false,"property","Person","elementType","relationship","otherLabels",List.of()));
+        expectedResult.add(MapUtil.map("other",List.of("Product"),"count",1L,"existence",false,"index",false,"label","Person","right",1L,"type","RELATIONSHIP","sample",null,"rightCount",1L,"leftCount",1L,"array",false,"left",1L,"unique",false,"property","BOUGHT","elementType","node","otherLabels",List.of()));
+        expectedResult.add(MapUtil.map("other",List.of(),"count",0L,"existence",false,"index",false,"label","Person","right",0L,"type","STRING","sample",null,"rightCount",0L,"leftCount",0L,"array",false,"left",0L,"unique",false,"property","name","elementType","node","otherLabels",List.of()));
+
+        TestUtil.testResult(db, "CALL apoc.meta.data.of('MATCH p = ()-[:BOUGHT]->() RETURN p')",
+                (r) -> {
+                    Set<Map<String, Object>> result = r.stream().collect(Collectors.toSet());
+                    assertEquals(expectedResult, result);
+                });
+    }
+
+    @Test
+    public void testMetaGraphOf() throws Exception {
+        db.executeTransactionally("CREATE (p:Person {name:'Tom Hanks'}), (m:Movie {title:'Forrest Gump'}), (pr:Product{name: 'Awesome Product'}), " +
+                "(p)-[:VIEWED]->(m), (p)-[:BOUGHT{quantity: 10}]->(pr)");
+
+        TestUtil.testResult(db, "CALL apoc.meta.graph.of('MATCH p = ()-[:BOUGHT]->() RETURN p')",
+                (r) -> {
+                    Map<String, Object> row = r.next();
+                    List<Node> nodes = (List<Node>) row.get("nodes");
+                    List<Relationship> relationships = (List<Relationship>) row.get("relationships");
+                    assertEquals(2,nodes.size());
+                    assertEquals(1,relationships.size());
+                    boolean hasLabels = nodes.stream().allMatch(n -> n.hasLabel(Label.label("Person")) || n.hasLabel(Label.label("Product")));
```

Comment:
maybe rewrite this assertions, as it would currently pass if you had 2 `:Person` nodes or 2 `:Product` nodes

Reply:
done!